Lib/zipfile/__init__.py

```diff
@@ -1817,6 +1817,11 @@ def write(self, filename, arcname=None,
         zinfo = ZipInfo.from_file(filename, arcname,
                                   strict_timestamps=self._strict_timestamps)
 
+        if isinstance(self.filename, os.PathLike) or isinstance(self.filename, bytes) \
+            or isinstance(self.filename, str):
+            if os.path.abspath(filename) == os.path.abspath(self.filename):
```

Comment:
I think we might want to use os.path.resolve() here, to resolve symlinks as well.

Reply:
Will make this fix later on today